java-symbol-solver-model/src/main/java/me/tomassetti/symbolsolver/model/typesystem/ReferenceTypeUsage.java

```diff
@@ -292,7 +292,8 @@ protected boolean compareConsideringTypeParameters(ReferenceTypeUsage other) {
         }
         if (this.getQualifiedName().equals(other.getQualifiedName())) {
             if (this.parameters().size() != other.parameters().size()) {
-                throw new IllegalStateException();
+                // one have the type not specified so they are compatible
+                return true;
```

Comment:
Can you push the branch and tell me the name? I would take it and apply to my branch to see how it affects the number of calls resolved